zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbMembershipState.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.authorization;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbCompositeKey;
+import io.camunda.zeebe.db.impl.DbEnumValue;
+import io.camunda.zeebe.db.impl.DbNil;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.mutable.MutableMembershipState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+public final class DbMembershipState implements MutableMembershipState {
+  private final EntityKeyAndRelationKey entityKeyAndRelationKey = new EntityKeyAndRelationKey();
+  private final RelationKeyAndEntityKey relationKeyAndEntityKey = new RelationKeyAndEntityKey();
+
+  private final ColumnFamily<EntityKeyAndRelationKey, DbNil> relationsByEntity;
+  private final ColumnFamily<RelationKeyAndEntityKey, DbNil> entitiesByRelation;
+
+  public DbMembershipState(
+      final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
+    relationsByEntity =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.RELATIONS_BY_ENTITY,
+            transactionContext,
+            entityKeyAndRelationKey,
+            DbNil.INSTANCE);
+
+    entitiesByRelation =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.ENTITIES_BY_RELATION,
+            transactionContext,
+            relationKeyAndEntityKey,
+            DbNil.INSTANCE);
+  }
+
+  @Override
+  public void insertRelation(
+      final EntityType entityType,
+      final String entityId,
+      final RelationType relationType,
+      final String relationId) {
+    entityKeyAndRelationKey.setAll(entityType, entityId, relationType, relationId);
```

Comment:
❓ This felt a little jarring to me, the CF etc is entityKey and relationKey but the values we use are `id`, should it really be entityIdAndRelationId?

Reply:
Here I chose `key` because it's literally the key in the kye-value database. Since all of these entities will no longer have a key, I think it's fine to keep it as is long term, even though it may look a bit confusing now because we know that they used to have keys in addition to ids.

Reply:
Ah I see, thank you for the context here, I was thinking too much about the entity key instead of the literal key in the database :)